activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -195,9 +196,20 @@ def insert(arel, name = nil, pk = nil, id_value = nil, sequence_name = nil, bind
       alias create insert
 
       # Executes the update statement and returns the number of rows affected.
-      def update(arel, name = nil, binds = [])
+      #
+      # Some adapters support the `returning` keyword argument which allows defining the return value of the method:
+      # `nil` is the default value and maintains default behavior. If an array of column names is passed then
+      # ActiveRecord::Result is returned from the method. The `rows` attribute of this returned object will
+      # contain the values of the specified columns for the updated row.
+      def update(arel, name = nil, binds = [], returning: nil)
         sql, binds = to_sql_and_binds(arel, binds)
-        exec_update(sql, name, binds)
+        result = exec_update(sql, name, binds, returning: returning)
+
+        if result.is_a?(ActiveRecord::Result) && returning.nil?
```

Comment:
I don't think I understood this conditional. It is that sometime we return `nil` and sometimes we return a result? If that is the case I'd prefer to keep `update` returning where it was and introduce another method to return the result for the adapter that needs it.